backend/application/syson-application/src/test/java/org/eclipse/syson/application/libraries/imports/SysONLibraryImportByReferenceTests.java

```diff
@@ -147,4 +157,53 @@ public void testImportingProjectSemanticDataHaveChanged(CapturedOutput capturedO
                     .hasSameElementsAs(libraryCopiedResource.getContents());
         }
     }
+
+    @Test
+    @DisplayName("The validation view of the project displays the same amount of validation rules after the import")
+    @SysONLibraryImportTestServer
+    public void testValidationAfterImport() {
+        ValidationEventInput validationEventInput = new ValidationEventInput(UUID.randomUUID(), this.projectEditingContext.getId(), "validation://");
+        var validationFlux = this.validationEventSubscriptionRunner.run(validationEventInput);
+
+        AtomicReference<Integer> diagnosticCount = new AtomicReference<>(0);
+
+        Consumer<Object> validationContentConsumer = payload -> Optional.of(payload)
+                .filter(ValidationRefreshedEventPayload.class::isInstance)
+                .map(ValidationRefreshedEventPayload.class::cast)
+                .map(ValidationRefreshedEventPayload::validation)
+                .ifPresentOrElse(validation -> {
+                    assertThat(validation).isNotNull();
+                    diagnosticCount.set(validation.getDiagnostics().size());
+                }, () -> fail("Missing validation"));
+
+        Runnable importLibrary = () -> {
+            this.importLibraryV1();
+        };
+
+        Consumer<Object> validationUpdatedContentConsumer = payload -> Optional.of(payload)
+                .filter(ValidationRefreshedEventPayload.class::isInstance)
+                .map(ValidationRefreshedEventPayload.class::cast)
+                .map(ValidationRefreshedEventPayload::validation)
+                .ifPresentOrElse(validation -> {
+                    assertThat(validation).isNotNull();
+                    assertThat(validation.getDiagnostics().size()).isEqualTo(diagnosticCount.get());
+                }, () -> fail("Missing validation"));
+
+        StepVerifier.create(validationFlux)
+                .consumeNextWith(validationContentConsumer)
+                .then(importLibrary)
+                .consumeNextWith(validationUpdatedContentConsumer)
+                .thenCancel()
+                .verify(Duration.ofSeconds(10));
+
+    }
+
+    private void importLibraryV1() {
```

Comment:
I moved this away from the `@BeforeEach` because I needed to check stuff _before_ the import.
In any case, I think this class should be reworked because it doesn't rely on the GraphQL APIs like the other integration tests (see for example line 207 `projectEditingContextEventProcessor.handle` which hooks into the event processor instead of a query runner).

This is not the scope of this PR though, so I adapted the code to fit my new test case.